src/components/Form/ProfileForm.tsx

```diff
@@ -215,9 +182,14 @@ function ProfileForm({ callback }: { callback?: () => void }) {
                   <div className='ms-auto'>
                     <button
                       disabled={aiLoading}
-                      onClick={e => {
+                      onClick={async e => {
                         e.preventDefault();
-                        generatePicture(setFieldValue);
+                        setAiLoading(true);
```

Comment:
move this part in a small function on top. Better if we want to optimize it and use useCallback